Explorer/Assets/DCL/PerformanceAndDiagnostics/Optimization/PerformanceBudgeting/IBudgetedDisposable.cs

```diff
@@ -0,0 +1,71 @@
+using Cysharp.Threading.Tasks;
+using System;
+using System.Collections.Generic;
+
+namespace DCL.Optimization.PerformanceBudgeting
+{
+    public interface IBudgetedDisposable
+    {
+        /// <summary>
+        /// Iterative dispose method. It allows to spread dispose load over frames to avoid spikes.
+        /// Each next call triggers internal dispose logic.
+        /// Synchronous replacement for IUniTaskAsyncDisposable when it's clear the dispose logic can be performed by chunks.
+        /// </summary>
+        IEnumerator<Unit> BudgetedDispose();
+    }
+
+    /// <summary>
+    /// Type represents empty value.
+    /// </summary>
+    public readonly struct Unit { }
```

Comment:
Call it Nothing or EmptyStruct. I'm guessing "unit" comes from some type theory or mathematics background, but you cannot assume other programmers know it.